model/map/src/main/java/org/keycloak/models/map/user/MapUserProvider.java

```diff
@@ -834,4 +827,35 @@ public boolean moveCredentialTo(RealmModel realm, UserModel user, String id, Str
     public void close() {
 
     }
+
+    private DefaultModelCriteria<UserModel> addSearchToModelCriteria(String value,
+            DefaultModelCriteria<UserModel> mcb) {
+
+        if (value.length() >= 2 && value.charAt(0) == '"' && value.charAt(value.length() - 1) == '"') {
+            // exact search
+            value = value.substring(1, value.length() - 1);
+        } else {
+            if (value.length() >= 2 && value.charAt(0) == '*' && value.charAt(value.length() - 1) == '*') {
+                // infix search
+                value = "%" + value.substring(1, value.length() - 1) + "%";
+            } else {
+                // default to prefix search
+                if (value.length() > 0 && value.charAt(value.length() - 1) == '*') {
+                    value = value.substring(0, value.length() - 1);
+                }
+                value += "%";
+
+                // workaround for KEYCLOAK-18969: Single wildcard (i.e "%") is currently not accepted.
+                if ("%".equals(value)) {
+                    value = "%%";
+                }
```

Comment:
I've tried it and it seems the issue is not reproducible on current main.

Reply:
Removed. The JpaUserProvider also doesn't include it, so it wouldnt make sense to have it only in this place anyway.